compiler/runtime/intern.c

```diff
@@ -249,14 +249,6 @@ void* SKIP_intern_shared(void* obj) {
   sk_stack3_t* st3 = &st3_holder;
   size_t nbr_pages = sk_get_nbr_pages(NULL);
   sk_cell_t* pages = sk_get_pages(nbr_pages);
-  int* large_pages = sk_malloc(sizeof(int) * nbr_pages);
```

Comment:
I see you're removing this because it's not read. Was it ever? Is it actually a bug that we're not reading it?

Reply:
It was read in a piece of code that I removed because it was dead as well.

I'm not sure why it was there, I supposed it was a remain from the past...

I see how `copy` benefits from large pages: it reuses them instead of copying them. But I don't see how `intern` could benefit from large pages.

Reply:
Looks like @skiplabsdaniel did something related in 5a3176f02a286efe122609c30132f6f7f9eb5648

Reply:
It was added by @pikatchu in the tiny commit 14f86be2a9b3ef127c48f0a3fed33813fce8a84a

Reply:
I think the goal was either:

1. to mark the large page as empty so that it can be reused, which kinda makes sense, but:
- to me, nothing guarantees there are no other pointers to the object in the large page, but I haven't looked at (hidden?) invariants on the usage of `intern` which could guarantee that;
- this was modifying only the local array of pages which doesn't survive `intern`.

2. or to make the binary search skip over the large pages that have already been treated, but I don't see the point. Did I miss something?

Reply:
Looks like it's probably just dead code from a failed experiment. Just thought I'd double check. :)

Reply:
Could also be left-over debugging code hunting for a memory leak.